wgpu-core/src/device/life.rs

```diff
@@ -355,7 +278,7 @@ impl LifetimeTracker {
             .active
             .iter_mut()
             .find(|a| a.index == last_submit_index)
-            .map(|a| &mut a.temp_resources);
+            .map(|a| &mut a.encoders.last_mut().unwrap().temp_resources);
```

Comment:
This was a little subtle for me. This needs a comment explaining that, because this resource's `last_submit_index` matches `a.index`, we know that `a` must have done something with the resource, so `a.encoders` must not be empty.

Reply:
Oh, this comment is on the third commit, "add `temp_resources`..."